cg/services/bcl_convert_metrics_service/parser.py

```diff
@@ -175,3 +167,55 @@ def has_undetermined_reads_in_lane(self, lane: int) -> bool:
             metrics=self.quality_metrics, sample_internal_id=UNDETERMINED, lane=lane
         )
         return bool(metrics)
+
+    @classmethod
+    def calculate_total_reads_for_metrics(cls, read_pair_count: int) -> int:
+        """Scale to read pair number up to single reads."""
+        return read_pair_count * SCALE_TO_READ_PAIRS
+
+    def get_aggregate_total_reads_for_metrics(self):
+        """Return the aggregate reads for the whole demux metrics."""
+        aggregate_read_pairs: int = self.get_aggregate_for_attribute(
+            metrics=self.demux_metrics, attr_name="read_pair_count"
+        )
+        return self.calculate_total_reads_for_metrics(read_pair_count=aggregate_read_pairs)
+
+    def get_aggregate_undetermined_reads_for_metrics(self) -> int:
+        """Calculate the total undetermined reads to the demux metrics."""
+        aggregate_undetermined_read_pairs: int = 0
+        for demux_metric in self.demux_metrics:
+            if demux_metric.sample_internal_id == UNDETERMINED:
+                aggregate_undetermined_read_pairs += demux_metric.read_pair_count
+        return self.calculate_total_reads_for_metrics(
+            read_pair_count=aggregate_undetermined_read_pairs
+        )
+
+    def get_mean_percent_q30_for_aggregated_metrics(self) -> float:
```

Comment:
Can we rename the "aggregate" functions to "for_flow_cell"? I feel like in some of these there are so many terms that one looses track of what the function does.
```suggestion
    def get_percent_q30_for_flow_cell(self) -> float:
```